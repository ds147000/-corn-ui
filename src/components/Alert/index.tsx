import React from 'react'
import { View } from '@tarojs/components'
import classNames from 'classnames'
import { ViewProps } from '../../types/View'
import Icon from '../Icon'

const Alter: React.FC<ViewProps> = ({ children, className, ...props }) => {
  const _class = classNames(className, 'xrk-alert', 'xrk-f')
  return (
    <View className={_class} {...props} >
      <Icon name="error" className="icon" />
      {children}
    </View>
  )
}

export default Alter
