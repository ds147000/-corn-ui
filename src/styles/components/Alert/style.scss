.xrk-alert {
  padding: 22px 30px;
  text-align: center;
  background: #ffeff1;
  font-size: 24px;
  font-weight: 400;
  color: hsl(0 , 100% , 65%);
  line-height: 36px;

  .icon {
    margin-right: 24px;
    font-size: 28px;
  }
}
