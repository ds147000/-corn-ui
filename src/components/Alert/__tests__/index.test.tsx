import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import Alert from '../index'

describe('comps/Alert', () => {
  test('基本样式', () => {
    const screen = render(
      <Alert>
        您发表的【商品名称】内容审核不通过，具体原因如下：文案文字，这里有可能有很多文字，允许多行显示
      </Alert>
    )
    screen.getByText('您发表的【商品名称】内容审核不通过，具体原因如下：文案文字，这里有可能有很多文字，允许多行显示')
    expect(screen).toMatchSnapshot()
  })


  test('事件', () => {
    const onClick = jest.fn()
    const screen = render(
      <Alert onClick={onClick} >
        您发表的【商品名称】内容审核不通过，具体原因如下：文案文字，这里有可能有很多文字，允许多行显示
      </Alert>
    )
    screen.getByText('您发表的【商品名称】内容审核不通过，具体原因如下：文案文字，这里有可能有很多文字，允许多行显示')
    fireEvent.click(screen.getByText('您发表的【商品名称】内容审核不通过，具体原因如下：文案文字，这里有可能有很多文字，允许多行显示'))
    expect(onClick).toHaveBeenCalledTimes(1)
    expect(screen).toMatchSnapshot()
  })
})
