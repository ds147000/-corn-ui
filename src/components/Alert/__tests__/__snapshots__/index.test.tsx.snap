// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comps/Alert 基本样式 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-alert xrk-f"
      >
        <i
          class="icon xrk-icon xrk-i-error"
        />
        您发表的【商品名称】内容审核不通过，具体原因如下：文案文字，这里有可能有很多文字，允许多行显示
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-alert xrk-f"
    >
      <i
        class="icon xrk-icon xrk-i-error"
      />
      您发表的【商品名称】内容审核不通过，具体原因如下：文案文字，这里有可能有很多文字，允许多行显示
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
